src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1045,6 +1049,28 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 	foreach(rangeTableCell, rangeTableList)
 	{
 		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
+		bool referenceTable = false;
+
+		if (IsDistributedTable(rangeTableEntry->relid))
```

Comment:
For instance, 
`RouterSelectTask()-> RouterSelectQuery() -> TargetShardIntervalsForSelect()` directly calls `DistributedTableCacheEntry()` on the relations.